lib/options/option_validator.rb

```diff
@@ -0,0 +1,81 @@
+# frozen_string_literal: true
+
+require_relative 'option'
+require_relative '../features/support/capabilities/devices'
+
+module Maze
+  # Validates command line options
+  class OptionValidator
+    # Validates all provided options
+    # @param options [Hash] Parsed command line options
+    def validate(options)
+      @errors = []
+      @options = options
+
+      # Common options
+      farm = options[Option::FARM]
+      if farm && !%w[bs local].include?(farm)
+        @errors << "--#{Maze::Option::FARM} must be either 'bs' or 'local' if provided"
+      end
+
+      begin
+        JSON.parse(options[Option::CAPABILITIES])
+      rescue JSON::ParserError
+        @errors << "--#{Maze::Option::CAPABILITIES} must be valid JSON"
+      end
+
+      # Farm specific options
+      validate_bs if farm == 'bs'
+      validate_local if farm == 'local'
+
+      @errors
+    end
+
+    # Validates BrowserStack options
+    def validate_bs
+      @errors << "--#{Maze::Option::APP} must be specified" if @options[Maze::Option::APP].nil?
```

Comment:
why use local variable storage instead of returning an array to `validate`? seems unnecessary to cache it when its only used in this one case and only called once.

Reply:
Changed to pass through local (to method) variables.